var/spack/repos/builtin/packages/py-entrypoints/package.py

```diff
@@ -25,15 +25,22 @@
 from spack import *
 
 
-class PyEntrypoints(Package):
+class PyEntrypoints(PythonPackage):
     """Discover and load entry points from installed packages."""
 
     homepage = "https://pypi.python.org/pypi/entrypoints"
-    url      = "https://files.pythonhosted.org/packages/f8/ad/0e77a853c745a15981ab51fa9a0cb4eca7a7a007b4c1970106ee6ba01e0c/entrypoints-0.2.2-py2.py3-none-any.whl"
+    url      = "https://pypi.python.org/packages/f8/ad/0e77a853c745a15981ab51fa9a0cb4eca7a7a007b4c1970106ee6ba01e0c/entrypoints-0.2.2-py2.py3-none-any.whl"
+
+    import_modules = ['entrypoints']
 
     version('0.2.2', '73bd7ce92c19b25dc5a20aff41be996a', expand=False)
 
+    depends_on('python@2.7:', type=('build', 'run'))
+
     depends_on('py-pip', type='build')
+    depends_on('py-configparser', when='^python@:2.8', type=('build', 'run'))
```

Comment:
Because:
```python
depends_on('py-configparser@3.5:', when='^python@2.7', type=('build', 'run'))
```
means it depends on `py-configparser` only when building with Python 2.7. _Exactly_ 2.7. Since the default is 2.7.13 (not 2.7), it doesn't depend on `py-configparser`. Normally this would be fine and the dependency would be ignored, but the concretizer is currently broken (see #2793) resulting in the bug you see. At least, that's my understanding of the problem. @tgamblin knows more about the concretizer than me.

Keep in mind that none of these bugs are by design, and we would like to see them fixed just as much as you. @tgamblin is working on a complete overhaul of the concretizer, so I expect to see this bug solved in the next release.